packages/helpers/src/phonebook.rs

```diff
@@ -0,0 +1,29 @@
+#[macro_export]
+macro_rules! get_contracts {
+    ($deps:expr, $factory_contract:expr, $($field_name:ident),*) => {
+        {
+            #[derive(Debug)]
+            struct Phonebook {
+                $(
+                    $field_name: String,
+                )*
+            }
+            let contracts = $deps
+                                .querier
+                                .query::<HashMap<String, String>>(&QueryRequest::Wasm(WasmQuery::Smart {
+                                    contract_addr: $factory_contract.to_string(),
+                                    msg: to_json_binary(&drop_staking_base::msg::factory::QueryMsg::Locate {
+                                        contracts: vec![$(stringify!($field_name).to_string()),*],
+                                    })?,
+                                }))?;
+
+            Phonebook {
+                $(
+                    $field_name: contracts.get(stringify!($field_name))
+                        .unwrap_or_else(|| panic!("Field {} not found in contracts", stringify!($field_name)))
```

Comment:
Can you use `Err()` instead of `panic!` here?

Reply:
this will lead to requirement of the error in all the places where this helper is used. on the other hand we must be pretty sure that contract address is already in the factory

Reply:
Btw, you can use `.expect("")` instead of `.unwrap_or_else(|| panic!(""))`